### DISCUSSION TITLE
Run in a script with pageObject

### DISCUSSION THREAD

**Discussion by DELEPaps** [webdriverio/webdriverio#8499] [Q&A]
i'm from reactjs and i'm new there, i'm just learning reactjs functional components.

basically i don't understand class in javascript.

I want to ask how to run webdriverio in script with page obejct
I have searched documentation and internet for a week but haven't found any enlightenment

> documentation run in a script
```
// index.ts
const { remote } = require('webdriverio');

// how to make browser global
(async () => {
    const browser = await remote({
        capabilities: {
            browserName: 'chrome'
        }
    })
    await browser.url('https://webdriver.io')
    await browser.deleteSession()
})()
```

> documentation page object pattern
```
// page.js
export default class Page {
  public open(path: string) {
    return browser.url(`https://the-internet.herokuapp.com/${path}`); // cant find name 'browser'
  }
}
```

Thanks in advance, and sorry my English is bad.

**Comment by erwinheitzman** [comment]
For an example on how to do this I have just created an example that can be found here: https://github.com/erwinheitzman/jest-webdriverio-standalone-boilerplate

Note: the variables `browser` and `driver` seem to be overwritten / reserved by webdriverio or another package, hence I used `chrome` in my example. You can use any other name but the two I just mentioned.

**Reply by DELEPaps** [reply]
I've looked at your script as well as wdio, then I tried several possibilities to modify mine, but all seem to fail.

is it possible to start wdio without a javascript testing tool? and access it globally?

**Reply by christian-bromann** [reply]
> but all seem to fail

You will have to tell us what did fail, otherwise it is impossible to help.

> is it possible to start wdio without a javascript testing tool? and access it globally?

I don't know what this means

**Reply by erwinheitzman** [reply]
EDIT: @DELEPaps yes this is possible but you will have to deal with all the things that test runners exist for, yourself. Like you are experiencing right now. I of course don't know your experience level with Javascript and NodeJS but I would never recommend anyone to not use a testrunner for the above mentioned reason. Test runners fulfill a purpose and that is to solve problems that people will run into when using pure Javascript.

For example, you will not have hooks so how will you control your flow?

Hope this helps.
